enterprise/frontend/src/metabase-enterprise/collections/components/CollectionAuthorityLevelIcon.tsx

```diff
@@ -0,0 +1,37 @@
+import React from "react";
+
+import Icon, { IconProps } from "metabase/components/Icon";
+import { color } from "metabase/lib/colors";
+
+import type { Collection } from "metabase-types/api";
+
+import { AUTHORITY_LEVELS } from "../constants";
+import { isRegularCollection } from "../utils";
+
+interface Props extends Omit<IconProps, "name" | "tooltip"> {
+  collection: Collection;
+
+  // check OFFICIAL_COLLECTION authority level definition
+  // https://github.com/metabase/metabase/blob/d0ab6c0e2361dccfbfe961d61e1066ec2faf6c40/enterprise/frontend/src/metabase-enterprise/collections/constants.js#L14
+  tooltip?: "default" | "belonging";
+}
+
+export function CollectionAuthorityLevelIcon({
+  collection,
+  tooltip = "default",
+  ...iconProps
+}: Props) {
+  if (isRegularCollection(collection)) {
+    return null;
+  }
+  const level = AUTHORITY_LEVELS[collection.authority_level as any];
```

Comment:
Can we avoid that cast? Will typing `AUTHORITY_LEVELS` as `Record<string, ...` help?

Reply:
@ranquild yeah, I also don't like that; the thing is that `authority_level` is `null | "official"` and TypeScript doesn't like `null` being a key of a record :/ Do you happen to know a better workaround for this?

Reply:
Yeah `null` cannot be the key of the object. In runtime it gets converted to string. So we need to do:
```
const AUTHORITY_LEVELS: Record<string, string> = {
  [String(regularCollection.type)]: ...
};
```

Reply:
And when you access it, `const level = AUTHORITY_LEVELS[String(collection.authority_level)]`

Reply:
Thanks! Let me try that 👍